File: practica/python/2.3.py
from msvcrt import getch

NumberMonth = int(input("Введіть номер місяця: "))

#case (використання словників)
print("Конструкція case (використання словників):", end = "\n\t")

case = {
    12: "Зима",
    1: "Зима",
    2: "Зима",
    3: "Весна",
    4: "Весна",
    5: "Весна",
    6: "Літо",
    7: "Літо",
    8: "Літо",
    9: "Осінь",
    10: "Осінь",
    11: "Осінь"
}

if(NumberMonth in case):
    print(case[NumberMonth])
else:
    print("Немає такого місяця")

#if
print("Конструкція if:", end = "\n\t")

if( (NumberMonth == 12) or (NumberMonth == 1) or (NumberMonth == 2) ):
    print("Зима")
    
elif( (NumberMonth >= 3) and (NumberMonth <= 5) ):
    print("Весна")

elif( (NumberMonth >= 6) and (NumberMonth <= 8) ):
    print("Літо")

elif( (NumberMonth >= 9) and (NumberMonth <= 11) ):
    print("Осінь")

else:
    print("Немає такого місяця")

getch()

File: practica/C++/2.3.cpp
#include <iostream>

#include <windows.h>
#include <conio.h>

using namespace std;

int main()
{
    SetConsoleCP(1251);
    SetConsoleOutputCP(1251);

    int NumberMonth;
    cout << "Введіть номер місяця: ";
    cin >> NumberMonth;

    //case
    cout << "Конструкція case:" << "\n" << "\t";
    switch (NumberMonth)
    {
        case 12:
        case 1:
        case 2:
            cout << "Зима" << "\n";
            break;
            
        case 3:
        case 4:
        case 5:
            cout << "Весна" << "\n";
            break;

        case 6:
        case 7:
        case 8:
            cout << "Літо" << "\n";
            break;

        case 9:
        case 10:
        case 11:
            cout << "Осінь" << "\n";
            break;

        default:
            cout << "Задане число не є місяцем" << "\n";
            break;
    }

    //if
    cout << "Конструкція if:" << "\n" << "\t";
    if( (NumberMonth == 12) || (NumberMonth == 1) || (NumberMonth == 2) ) 
    {
        cout << "Зима" << "\n";
    }
    else if( (NumberMonth >= 3) && (NumberMonth <= 5) )
    {
        cout << "Весна" << "\n";
    }
    else if ( (NumberMonth >= 6) && (NumberMonth <= 8) )
    {
        cout << "Літо" << "\n";    
    }
    else if( (NumberMonth >= 9) && (NumberMonth <= 11) )
    {
        cout << "Осінь" << "\n";
    }
    else
    {
        cout << "Задане число не є місяцем" << "\n";
    }

    getch();

    return 0;
}